blogp3.md

```diff
@@ -0,0 +1,434 @@
+
+Once we get the token, we set our state variable `token` and make a API call using the token to get the user's display name and id. To test that this actually works, let's render a "Logged in as {display_name}" statement. This will change later.
+
+```
+import React from 'react';
+
+class UploadFace extends React.Component {
+  constructor(){
+    super()
+    
+    this.state = {
+      token: "",
+      userInfo: []
+    }
+  }
+
+  //calls getHashParams and uses token to get user info
+  componentDidMount(){
+    const params = this.getHashParams();
+    this.setState({token: params.access_token})
+  
+    fetch('https://api.spotify.com/v1/me',{
+      method: 'GET',
+      headers: { 'Authorization' : 'Bearer ' + params.access_token}
+    }).then(resp => resp.json()).then(data => this.setState({userInfo: [data.display_name, data.id]}))
+    
+  }
+
+  //this parses the query string and extracts the token
+  getHashParams() {
+    console.log('in hash params')
+    var hashParams = {};
+    var e, r = /([^&;=]+)=?([^&;]*)/g,
+        q = window.location.hash.substring(1);
+    while ( e = r.exec(q)) {
+       hashParams[e[1]] = decodeURIComponent(e[2]);
+    }
+    return hashParams;
+  }
+
+  render(){
+    <h3>Logged in as {this.state.userInfo[0]}</h3>
+  }
+}
+
+export default UploadFace;
+```
+
+Don't run this just yet- we have some routing to do.
+
+#### Step 3- Adding Routing
+
+Go back into your App.js file. In your terminal run `npm install react-router-dom` . We'll be using react-router-dom to route our function, meaning that we'll be able to display different pages depending on the url.
+
+Copy paste this code:
+
+Notice that I'm using `BrowserRouter`(saved as Router) and `Route` from the react-router-dom library. I'm surrounding all the components in my return statement with `<Router><Router/>`, which will allow me to specifiy actual Routes.
+
+```
+import React from 'react';
+import './App.css';
+import UploadFace from './components/UploadFace'
+import Home from './components/Home'
+import {BrowserRouter as Router, Route} from 'react-router-dom';
+
+class App extends React.Component {
+
+  render() {
+    return (
+      <Router>
+        <div className="App">
+          <Nav />
+          <Route path="/face" component={UploadFace}/>
+          <Route path="/" exact component={Home}/>
+        </div>
+      </Router>  
+    );
+  }
+}
+
+export default App;
+```
+
+These lines:
+
+```
+<Route path="/face" component={UploadFace}/>
+<Route path="/" exact component={Home}/>
+```
+
+are doing the actual routing. The path indicates the url that the user needs to go to for the page to render, and the component is the component that should be rendered at that specific url. Notice that there's a `<Nav />` component that isn't surrounded by route- this means that it will alway be rendered regardless of url. This is the navbar at the top of the page- you can customize this however you want, or eliminate it entirely.
```

Comment:
```suggestion
are doing the actual routing. The path indicates the url that the user needs to go to for the page to render, and the component is the component that should be rendered at that specific url. Notice that there's a `<Nav />` component that isn't surrounded by route- this means that it will always be rendered regardless of the given url. This is the navbar at the top of the page- you can customize this however you want, or eliminate it.
```